data_steward/cdr_cleaner/cleaning_rules/sandbox_and_remove_pids.py

```diff
@@ -1,120 +1,148 @@
-    :param project_id: b1 project_id
-    :param dataset_id: bq dataset_id
-    :param lookup_table: name of the lookup table
-    :return: list of select statements that will truncate the existing tables with clean data
-    """
-    person_tables_list = get_tables_with_person_id(project_id, dataset_id)
-    queries_list = []
-
-    for table in person_tables_list:
-        delete_queries = CLEAN_QUERY.format(project=project_id,
-                                            dataset=dataset_id,
-                                            table=table,
-                                            sandbox_dataset=sandbox_dataset_id,
-                                            lookup_table=lookup_table)
-        queries_list.append({
-            clean_consts.QUERY: delete_queries,
-            clean_consts.DESTINATION_TABLE: table,
-            clean_consts.DESTINATION_DATASET: dataset_id,
-            clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
-        })
-
-    return queries_list
+    def __init__(self,
+                 issue_numbers=None,
+                 description: str = None,
+                 affected_datasets=None,
+                 project_id: str = None,
+                 dataset_id: str = None,
+                 sandbox_dataset_id: str = None,
+                 depends_on=None,
+                 affected_tables=None,
+                 table_namer: str = None):
+        """
+        Initialize the class with proper information.
+        """
+
+        desc = (
+            'Abstract class to sandbox and remove participant records from the '
+            'given dataset.  This is a class that can be extended for other rules.'
+        )
+        description = description if description else desc
+
+        super().__init__(issue_numbers=issue_numbers,
+                         description=description,
+                         affected_datasets=affected_datasets,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         affected_tables=affected_tables,
+                         depends_on=depends_on,
+                         table_namer=table_namer)
+
+        self.person_table_list = []
+
+    def setup_rule(self, client: BigQueryClient):
+
+        # def get_tables_with_person_id(self, client: BigQueryClient):
+        """
+        Get list of tables that have a person_id column, excluding mapping tables
+        """
+        if self.person_table_list:
+            return self.person_table_list
+
+        person_table_query = PERSON_TABLE_QUERY.render(project=self.project_id,
+                                                       dataset=self.dataset_id)
+        person_tables = client.query(person_table_query).result()
+        # utils.bq.query(person_table_query).get(
+        #     TABLE_NAME_COLUMN).to_list()
+        # exclude mapping tables from list, to be removed after all cleaning rules
```

Comment:
nit: comments can be removed
```suggestion
        """
        Get list of tables that have a person_id column, excluding mapping tables
        """
        if self.person_table_list:
            return self.person_table_list

        person_table_query = PERSON_TABLE_QUERY.render(project=self.project_id,
                                                       dataset=self.dataset_id)
        person_tables = client.query(person_table_query).result()
```